si2.api/Controllers/InstitutionsController.cs

```diff
@@ -92,8 +92,9 @@ public async Task<ActionResult> UpdateInstitution([FromRoute]Guid id, [FromBody]
                 return NotFound();
 
             var institutionToReturn = await _institutionService.UpdateInstitutionAsync(id, updateInstitutionDto, ct);
+            
             if (institutionToReturn == null)
-                return BadRequest();
+                return BadRequest("ParentId cannot be equal to the Institution's Id !");
```

Comment:
this message will appear whenever the institutionToReturn is null even if it's error other than the parent_id == id, you can just put this here in the controller: if (id == updateInstitutionDto.ParentId){  return BadRequest("ParentId cannot be equal to the Institution's Id !";  } else return { return BadRequest();}